examples/eks-cluster-with-argocd/main.tf

```diff
@@ -0,0 +1,184 @@
+terraform {
+  required_version = ">= 1.0.1"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.66.0"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.6.1"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = ">= 2.4.1"
+    }
+  }
+
+  backend "local" {
+    path = "local_tf_state/terraform-main.tfstate"
+  }
+}
+
+data "aws_region" "current" {}
+
+data "aws_availability_zones" "available" {}
+
+data "aws_eks_cluster" "cluster" {
+  name = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+}
+
+provider "aws" {
+  region = data.aws_region.current.id
+  alias  = "default"
+}
+
+provider "kubernetes" {
+  experiments {
+    manifest_resource = true
+  }
+  host                   = data.aws_eks_cluster.cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster.token
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.cluster.endpoint
+    token                  = data.aws_eks_cluster_auth.cluster.token
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  }
+}
+
+locals {
+  tenant      = "aws001"  # AWS account name or unique id for tenant
+  environment = "preprod" # Environment area eg., preprod or prod
+  zone        = "dever"   # Environment with in one sub_tenant or business unit
+
+  kubernetes_version = "1.21"
+
+  vpc_cidr       = "10.0.0.0/16"
+  vpc_name       = join("-", [local.tenant, local.environment, local.zone, "vpc"])
+  eks_cluster_id = join("-", [local.tenant, local.environment, local.zone, "eks"])
+
+  terraform_version = "Terraform v1.0.1"
+
+  #---------------------------------------------------------------
+  # ARGOCD ADD-ON APPLICATION
+  #---------------------------------------------------------------
+
+  addon_application = {
+    path               = "chart"
+    repo_url           = "git@github.com:aws-samples/ssp-eks-add-ons.git"
+    add_on_application = true
+  }
+
+  #---------------------------------------------------------------
+  # ARGOCD WORKLOAD APPLICATION
+  #---------------------------------------------------------------
+
+  workload_application = {
+    path               = "envs/dev"
+    repo_url           = "git@github.com:aws-samples/ssp-eks-workloads.git"
```

Comment:
We may nee dot change this https url to avoid SSH issue. We can document how users can you SSH as well by passing the key